rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/serialization/ExportNodeSourceToCatalogServlet.java

```diff
@@ -0,0 +1,133 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Servlet invoked when a node source is exported to the catalog. The request
+ * content is expected to be in JSON format, containing the configuration of a
+ * node source.
+ */
+public class ExportNodeSourceToCatalogServlet extends HttpServlet {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExportNodeSourceToCatalogServlet.class);
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) {
+        upload(request, response);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+        upload(request, response);
+    }
+
+    private void upload(HttpServletRequest request, HttpServletResponse response) {
+        response.setContentType("text/html");
+        try (CatalogObjectAction catalogObjectAction = buildCatalogObjetAction(request)) {
+            CatalogRequestBuilder catalogRequestBuilder = new CatalogRequestBuilder(catalogObjectAction);
+            String requestUri = catalogRequestBuilder.build();
+            LOGGER.info("Post node source to catalog with URI: " + requestUri);
+            try (CloseableHttpResponse httpResponse = catalogRequestBuilder.postNodeSourceRequestToCatalog(catalogObjectAction.getSessionId(),
+                                                                                                           requestUri,
+                                                                                                           response)) {
+                new BasicResponseHandler().handleResponse(httpResponse);
+            } catch (HttpResponseException e) {
+                logErrorAndWriteResponseToClient(e, response);
+            }
+        } catch (Exception e) {
+            logErrorAndWriteResponseToClient(e, response);
+        }
+    }
+
+    private CatalogObjectAction buildCatalogObjetAction(HttpServletRequest request) throws Exception {
+        CatalogObjectAction catalogObjectAction = new CatalogObjectAction();
+        for (FileItem formItem : new ServletRequestTransformer().getFormItems(request)) {
+            if (formItem.isFormField()) {
+                String fieldValue = formItem.getString();
+                switch (formItem.getFieldName()) {
+                    case SESSION_ID:
+                        catalogObjectAction.setSessionId(fieldValue);
+                        break;
+                    case BUCKET_NAME_PARAM:
+                        catalogObjectAction.setBucketName(fieldValue);
+                        break;
+                    case NAME_PARAM:
+                        catalogObjectAction.setNodeSourceName(fieldValue);
+                        break;
+                    case FILE_CONTENT_PARAM:
+                        File nodeSourceJsonFile = File.createTempFile("node-source-configuration", ".json");
+                        new PrintWriter(nodeSourceJsonFile).write(fieldValue);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "PrintWriter" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)